src/coins.cpp

```diff
@@ -283,7 +283,7 @@ bool CCoinsViewCache::Sync()
 void CCoinsViewCache::Uncache(const COutPoint& hash)
 {
     CCoinsMap::iterator it = cacheCoins.find(hash);
-    if (it != cacheCoins.end() && it->second.flags == 0) {
+    if (it != cacheCoins.end() && !it->second.IsDirty() && !it->second.IsFresh()) {
```

Comment:
I think this is right, just wanted to make a note that I believe @mzumsande earlier [comment](https://github.com/bitcoin/bitcoin/pull/28280#discussion_r1655477509) is relevant here. Namely, that it's not necessary to check for freshness here if we cannot have fresh-but-not-dirty entries, and even if we could, I'm not sure why we wouldn't remove them from the cache in this instance. But I think this should be addressed by any revival of #18746.

Reply:
Done in #30673 30673